internal/engine/upper_test.go

```diff
@@ -2450,32 +2450,35 @@ func TestSetAnalyticsOpt(t *testing.T) {
 	defer f.TearDown()
 
 	opt := func(ia InitAction) InitAction {
-		ia.AnalyticsOpt = analytics.OptIn
+		ia.AnalyticsOpt = analytics.OptDefault
 		return ia
 	}
 
 	f.Start([]model.Manifest{}, true, opt)
-	f.store.Dispatch(store.AnalyticsOptAction{Opt: analytics.OptOut})
-	f.WaitUntil("opted out", func(state store.EngineState) bool {
-		return state.AnalyticsOpt == analytics.OptOut
-	})
-
-	// if we don't wait for 1 here, it's possible the state flips to out and back to in before the subscriber sees it,
-	// and we end up with no events
-	f.opter.waitUntilCount(t, 1)
```

Comment:
is this no longer a concern?

Reply:
it is, but I was kinda on the fence about whether it was worth the space

added it back with better wording